polaris-react/src/components/Modal/components/Header/Header.tsx

```diff
@@ -24,13 +24,21 @@ export function Header({
 }: HeaderProps) {
   const {polarisSummerEditions2023} = useFeatures();
 
+  const headerPaddingInline = polarisSummerEditions2023 ? '4' : '5';
+  const headerPaddingBlock = '4';
+
   if (titleHidden || !children) {
-    return (
+    return polarisSummerEditions2023 ? (
       <Box
         position="absolute"
-        insetInlineEnd={polarisSummerEditions2023 ? '1' : '0'}
+        insetInlineEnd={headerPaddingInline}
+        insetBlockStart={headerPaddingBlock}
         zIndex="1"
       >
+        <CloseButton titleHidden={titleHidden} onClick={onClose} />
+      </Box>
```

Comment:
No idea why the old/below version has a `<HorizontalStack>` wrapping a single element - it does literally nothing 🤷 

So... I removed it.

Reply:
Not sure why this done in this instance, but wrapping single (or groups of) elements in a nested `HorizontalStack` is usually done to `align="end"` when inside of a `Box` with width of 100%.

Reply:
:oh-no: we're using `<HorizontalStack>` as a flex container? Conceptually, that's not a "Stack", but coincidentally due to implementation I can see how it's being leveraged for that because the flex container takes up 100% width by default. Feels like we need just a `<Flex>` primitive for these kinds of layout situations.